tools/alpine/detect-arch-var

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env sh
```

Comment:
Sure, of course! But interesting, for the 'ARG' and ‘FROM’ feature supported by the newer version of Docker, seems there's a potential bug for the DOCKER_CONTENT_TRUST setting:
$export DOCKER_CONTENT_TRUST=1
$docker build --no-cache --build-arg BASE=alpine:3.6 -t alpine:build .
You will get below message:
Sending build context to Docker daemon 

error during connect: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.31/build?buildargs=%7B%22BASE%22%3A%22alpine%3A3.6%22%7D&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=Dockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=default&nocache=1&rm=1&shmsize=0&t=alpine%3Abuild&target=&ulimits=null: invalid reference format: repository name must be lowercase

@justincormack  @rn  please help double check this, if it's done, my PR will be ready quickly to push...